### PR TITLE
New Prebid1.0 adapter for SpotX

### PR DESCRIPTION
**Pull request by strider820**
## Type of change

- [ ] Bugfix
- [ ] Feature
- [x] New bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change
Creating a new SpotX Bidder Adapter

- test parameters for validating bids
```
{
  bidder: 'spotx',
  params: {
    channel_id: '85394',
    ad_unit: 'outstream',
    outstream_options: {
      slot: 'div-gpt-ad-123445678-1',
      content_width:  '300',
      content_height: '250',
    }
  }
}
```

Be sure to test the integration with your adserver using the [Hello World](/integrationExamples/gpt/hello_world.html) sample page.

- jhernandez@spotx.tv
- [x] official adapter submission

For any changes that affect user-facing APIs or example code documented on http://prebid.org, please provide:

- https://github.com/prebid/prebid.github.io/pull/1110

### CODE REVIEW

File: modules/spotxBidAdapter.js [hunk 1]
```diff
@@ -0,0 +1,417 @@
+import * as utils from 'src/utils';
+import { Renderer } from 'src/Renderer';
+import { registerBidder } from 'src/adapters/bidderFactory';
+import { VIDEO } from 'src/mediaTypes';
```
Comment (robertrmartinez) [inline modules/spotxBidAdapter.js]: Imports like these need to be relative after `1.38.0`

See: 

so add `../` before each `src`

File: modules/spotxBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,417 @@
...
+
+      if (utils.getBidIdParameter('number_of_ads', bid.params)) {
+        requestPayload['ext']['number_of_ads'] = utils.getBidIdParameter('number_of_ads', bid.params);
+      }
+
+      const userExt = {};
+
+      // Add GDPR flag and consent string
+      if (bidderRequest && bidderRequest.gdprConsent) {
+        userExt.consent = bidderRequest.gdprConsent.consentString;
```
Comment (robertrmartinez) [inline modules/spotxBidAdapter.js]: If a pub enables the consent modul, but for some reason the consent information cannot be found,
then prebid passes an object like so to the bidders:

```javascript
{
   gdprConsent: {
      consentString: undefined
      gdprApplies: undefined
      vendorData: undefined
  }
}
```

So technically your `userExt` would be:

```javascript
{
   consent: undefined
}
```

at this point, but it gets pruned before the request eventually gets sent by prebid so no big deal, just though I would make aware that Prebid passes those undefined values in some instances!

File: test/spec/modules/spotxBidAdapter_spec.js [hunk 3]
```diff
@@ -0,0 +1,379 @@
+import {expect} from 'chai';
+import adapterManager from 'src/adapterManager';
+import {spec, outstreamRender} from 'modules/spotxBidAdapter';
+import {parse as parseQuery} from 'querystring';
+import {newBidder} from 'src/adapters/bidderFactory';
+import {userSync} from 'src/userSync';
+import {config} from 'src/config';
+import * as utils from 'src/utils';
+import find from 'core-js/library/fn/array/find';
+
```
Comment (robertrmartinez) [inline test/spec/modules/spotxBidAdapter_spec.js]: Almost all of these imports are never used besides `expect` and `spec`

Can you remove the unused ones?


Both `CONSTANTS` as well as `INTEGRATION` below here are not used either.

Let me know if I missed something and there is a reason for keeping them!

File: modules/spotxBidAdapter.js (hunk 2 above)
Comment (strider820) [reply modules/spotxBidAdapter.js]: Our ad server should handle undefined in those requests anyways, but that's good to know!

File: test/spec/modules/spotxBidAdapter_spec.js (hunk 3 above)
Comment (strider820) [reply test/spec/modules/spotxBidAdapter_spec.js]: Updated. I went ahead and wrote a few more tests while I was in there to bump that coverage up a bit.

Comment (robertrmartinez) [reply test/spec/modules/spotxBidAdapter_spec.js]: Awesome!! Thanks for that.

**Pull request merged**